packages/core/src/context.ts

```diff
@@ -21,15 +21,15 @@ export interface ContextProviderProps<T = unknown> {
 export function useContext<T>(context: ComponentContext<T>): T | undefined {
   // context must come from a parent
   let current = getContext();
-  while (current) {
```

Comment:
These changes shouldn't be necessary (or if they are, are masking some other bug I suspect). Can you say more about why this is needed?

Reply:
Ahh yea, I was using Alloy in typespec incorrectly, and the errors were pointing here. Just turns out it wasn't actually initing the context properly, forgot to revert this back.